app/src/main/java/org/mozilla/focus/widget/BrowserNavigationOverlay.kt

```diff
@@ -105,6 +105,13 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         autocompleteFilter.load(context.applicationContext)
         setOnFilterListener { searchText, view -> autocompleteFilter.onFilter(searchText, view) }
 
+        // #548: Get the current URL when we leave resign focus
+        setOnFocusChangeListener { _, hasFocus ->
+            if (!hasFocus) {
+                setText(navigationStateProvider?.getCurrentUrl())
```

Comment:
This unsets any input the user has entered when using the url bar – I don't like that (and desktop doesn't do that). I think we should just remove this OnFocusChangeListener entirely and just prevent the url bar from changing if the user already has focus.